testsuite/openshift/objects/metrics.py

```diff
@@ -0,0 +1,92 @@
+"""Simple client for the OpenShift metrics"""
+import time
+from datetime import datetime, timedelta
+from urllib.parse import urljoin
+from math import ceil
+
+import httpx
+
+
+# default scraping interval in seconds
+METRICS_SCRAPE_INTERVAL = 30
+
+
+def _params(key: str = "", labels: dict[str, str] = None) -> dict[str, str]:
+    """Generate metrics query parameters based on key and labels"""
+    if not labels:
+        return {"query": key}
+    # pylint: disable=consider-using-f-string
+    return {"query": "%s{%s}" % (key, ",".join(f"{k}='{v}'" for k, v in labels.items()))}
+
+
+class Metrics:
+    """Interface to the exposed Prometheus metrics by OpenShift"""
+
+    def __init__(self, endpoint: str, token: str, namespace: str = None):
+        self.endpoint = endpoint
+        self.token = token
+        self.headers = {"Authorization": f"Bearer {self.token}"}
+        self.namespace = namespace
+
+    def _request(self, path, **kwargs):
+        """Make a request to Prometheus API"""
+        url = urljoin(self.endpoint, path)
+        return httpx.get(url, verify=False, headers=self.headers, **kwargs)
+
+    def get_targets(self) -> dict:
+        """Get active metric targets information"""
+
+        params = {"state": "active"}
+
+        response = self._request("/api/v1/targets", params=params)
+        response.raise_for_status()
+        return response.json()["data"]["activeTargets"]
+
+    def get_metrics(self, key: str = "", labels: dict[str, str] = None) -> list:
+        """Get metrics by key or labels"""
+        if self.namespace:
+            labels = labels or {}
+            labels.setdefault("namespace", self.namespace)
+
+        params = _params(key, labels)
+
+        response = self._request("/api/v1/query", params=params)
+        response.raise_for_status()
+        return response.json()["data"]["result"]
+
+    @staticmethod
+    def get_metrics_keys(metrics: list):
+        """Return set of given metrics names"""
+        return {m["metric"]["__name__"] for m in metrics}
+
+    def wait_for_scrape(self, target_service: str, metrics_path: str = "/metrics", scrape_after: datetime = None):
```

Comment:
This method seems too long. Isn't there a better way how to way for scrape? My suggestion (not sure if it will work) is to see when the last scrape was made and wait until the time updates, through something like this https://groups.google.com/g/prometheus-users/c/vzbKGoJvzA8 ?

Reply:
I'll look into this. But I don't think that method is that long when you think about all input parameters that should be respected. It is an optimized way. At the beginning this method was just a `time.wait` one-liner, but in this case it will be waiting 30 seconds for every scrape (which won't be that time consuming for now, as method for scrape waiting is called only 2 times).